ngraph/core/src/op/rnn_sequence.cpp

```diff
@@ -0,0 +1,175 @@
+//*****************************************************************************
+// Copyright 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/rnn_sequence.hpp"
+#include "ngraph/op/util/recurrent_sequence.hpp"
+#include "ngraph/opsets/opset4.hpp"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+using namespace std;
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(op::v5::RNNSequence, "RNNSequence", 4);
+
+op::v5::RNNSequence::RNNSequence()
+    : m_direction(op::RecurrentSequenceDirection::FORWARD)
+{
+}
+
+op::v5::RNNSequence::RNNSequence(const Output<Node>& X,
+                                 const Output<Node>& H_t,
+                                 const Output<Node>& sequence_lengths,
+                                 const Output<Node>& W,
+                                 const Output<Node>& R,
+                                 const Output<Node>& B,
+                                 std::size_t hidden_size,
+                                 op::RecurrentSequenceDirection direction,
+                                 const std::vector<std::string>& activations,
+                                 const std::vector<float>& activations_alpha,
+                                 const std::vector<float>& activations_beta,
+                                 float clip)
+    : RNNCellBase({X, H_t, sequence_lengths, W, R, B},
+                  hidden_size,
+                  clip,
+                  activations,
+                  activations_alpha,
+                  activations_beta)
+    , m_direction(direction)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v5::RNNSequence::validate_and_infer_types()
+{
+    for (const auto& input : inputs())
+    {
+        if (input.get_partial_shape().rank().is_dynamic())
```

Comment:
what if the rank is known but shape in partial? further `get_shape()` will fail

Reply:
fixed